packages/tpetra/core/src/Tpetra_BlockCrsMatrix_Helpers_def.hpp

```diff
@@ -326,116 +337,114 @@ namespace Tpetra {
       const map_type &pointRangeMap = *(pointMatrix.getRangeMap());
       RCP<const map_type> meshRangeMap = createMeshMap<LO,GO,Node>(blockSize, pointRangeMap);
 
-      // Use graph ctor that provides column map and upper bound on nonzeros per row.
-      // We can use static profile because the point graph should have at least as many entries per
-      // row as the mesh graph.
-      RCP<crs_graph_type> meshCrsGraph = rcp(new crs_graph_type(meshRowMap, meshColMap,
-                                                 pointMatrix.getGlobalMaxNumRowEntries()));
-      // Fill the graph by walking through the matrix.  For each mesh row, we query the collection of point
-      // rows associated with it. The point column ids are converted to mesh column ids and put into an array.
-      // As each point row collection is finished, the mesh column ids are sorted, made unique, and inserted
-      // into the mesh graph.
-      typename crs_matrix_type::local_inds_host_view_type pointColInds;
-      typename crs_matrix_type::values_host_view_type pointVals;
-      Array<GO> meshColGids;
-      meshColGids.reserve(pointMatrix.getGlobalMaxNumRowEntries());
-
-      //again, I assume that point GIDs associated with a mesh GID are consecutive.
-      //if they are not, this will break!!
-      GO indexBase = pointColMap.getIndexBase();
-      for (size_t i=0; i<pointMatrix.getLocalNumRows()/blockSize; i++) {
-        for (int j=0; j<blockSize; ++j) {
-          LO rowLid = i*blockSize+j;
-          pointMatrix.getLocalRowView(rowLid,pointColInds,pointVals); //TODO optimization: Since I don't care about values,
-                                                                      //TODO I should use the graph instead.
-          for (size_t k=0; k<pointColInds.size(); ++k) {
-            GO meshColInd = (pointColMap.getGlobalElement(pointColInds[k]) - indexBase) / blockSize + indexBase;
-            if (meshColMap->getLocalElement(meshColInd) == Teuchos::OrdinalTraits<GO>::invalid()) {
-              std::ostringstream oss;
-              oss<< "["<<i<<"] ERROR: meshColId "<< meshColInd <<" is not in the column map.  Correspnds to pointColId = "<<pointColInds[k]<<std::endl;
-              throw std::runtime_error(oss.str());
-            }
+      RCP<crs_graph_type> meshCrsGraph;
+
+      const offset_type bs2 = blockSize * blockSize;
```

Comment:
I know that "all entries associated with a block are present" is a precondition, but it wouldn't hurt to check that pointMatrix's local nnz is divisible by bs2 in this function